app/repository.py

```diff
@@ -1,72 +1,43 @@
-from sqlalchemy.exc import SQLAlchemyError
-from app.database import Base, engine, SessionLocal
-from dataclasses import dataclass
+from app.database import database
 
 
-# creates database
-Base.metadata.create_all(engine)
-
-
-@dataclass
-class Err:
-    error: str
-
-
-NO_OBJECT_ERROR = "No such object!"
-
-
-def get_session():
-    session = SessionLocal()
-    try:
-        yield session
-    finally:
-        session.close()
+class NotFoundError(Exception):
+    pass
 
 
 class SQLiteRepository:
     @staticmethod
-    def fetch_all(model, session):
-        try:
-            data = session.query(model).all()
-            return data
-        except SQLAlchemyError as err:
-            return Err(str(err))
+    async def fetch_all(table):
+        query = table.select()
+        return await database.fetch_all(query)
 
     @staticmethod
-    def fetch_one(object_id, session, model):
-        try:
-            data = session.query(model).get(object_id)
-            if not data:
-                return Err(NO_OBJECT_ERROR)
-            else:
-                return data
-        except SQLAlchemyError as err:
-            return Err(str(err))
+    async def fetch_one(table, object_id):
+        query = table.select().where(table.c.id == object_id)
+        data = await database.fetch_one(query)
+        if not data:
+            return NotFoundError("No such object.")
+        return data
 
     @staticmethod
-    def _commit_changes(session):
-        try:
-            session.commit()
-        except SQLAlchemyError as err:
-            return Err(str(err))
+    async def add_values(table, values):
+        query = table.insert().values(values)
+        record_id = await database.execute(query)
+        query = table.select().where(table.c.id == record_id)
+        data = await database.fetch_one(query)
+        return data
```

Comment:
2 queries, no exception handling

Reply:
True, have an idea how to deal with the 2 queries but not sure what exception handling we need if fastAPI deals with HTTP 500 on its own. Shall we still have try/catch block here and something like 'except SQLAlchemyError as err'? But then in main.py we would need to catch this and return 500?

Reply:
If we decide that we don't need exception handling in the infrastructure layer because the app can handle it, what if we change our app layer and it can't take it out of the box anymore? We are handling them in other methods it's not consistent in my opinion.